src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx

```diff
@@ -0,0 +1,225 @@
+    Object.values(rotationData.rotations).forEach((rotation) => {
+      rotation.categories.forEach((category) => {
+        category.tasks.forEach((task) => {
+          if (task.isCompleted) {
+            initialChecked[task.id] = true;
+          }
+        });
+      });
+    });
+    setIsChecked(initialChecked);
+  }, [rotationData]);
+
+  const handleCheckboxChange = (taskId: string) => {
+    setIsChecked((prevChecked) => ({
+      ...prevChecked,
+      [taskId]: !prevChecked[taskId],
+    }));
+  };
+
+  const handleAllCheckedChange = (categoryId: string) => {
+    // Get all task IDs for the category
+    const taskIds = Object.values(rotationData.rotations).flatMap(
+      (rotation) =>
+        rotation.categories
+          .find((category) => category.id === categoryId)
+          ?.tasks.map((task) => task.id) || [],
+    );
+
+    // Check if all tasks in the category are checked
+    const allChecked = taskIds.every((taskId) => isChecked[taskId]);
+
+    // Create a new checked state with toggled values
+    const newChecked = { ...isChecked };
+    taskIds.forEach((taskId) => {
+      newChecked[taskId] = !allChecked;
+    });
+
+    // Update the checked state
+    setIsChecked(newChecked);
+  };
+
+  const { handleSubmit } = useForm();
+
+  const onSubmit = async () => {
+    setIsLoading(true);
+
+    // Update the tasks
+    const updatedTasks = Object.values(rotationData.rotations)
+      .flatMap((rotation) =>
+        rotation.categories.flatMap((category) => category.tasks),
+      )
+      .filter((task) => isChecked[task.id])
+      .map((task) => ({
+        ...task,
+        isCompleted: isChecked[task.id] || false,
+      }));
+    console.log('updated', updatedTasks);
+
+    // Submit the form data
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    setTimeout(() => {
+      setIsLoading(false);
+      toast({
+        variant: 'default',
+        description: TOAST_TEXTS.success,
+      });
+    }, 1000);
+  };
+
+  return (
+    <>
+      {isLoading && <LoadingSpinner isLoading={true} />}
+      <Accordion
+        type="single"
+        collapsible
+        className="w-full"
+        defaultValue={`item-1`}
+      >
+        <form onSubmit={handleSubmit(onSubmit)}>
+          {Object.entries(rotationData.rotations).map(([key, rotation]) => {
+            const numericKey = parseInt(key);
+            return (
+              <AccordionItem value={`item-${key}`} key={key}>
+                <AccordionTrigger>
+                  {rotation.startDate} - {rotation.endDate}
+                </AccordionTrigger>
+                <AccordionContent
+                  className={'bg-primary-lightest p-0 rounded-none'}
+                  asChild
+                >
+                  {rotation.categories.length === 0 ? (
+                    <p className="p-4 text-center">No task</p>
+                  ) : (
+                    rotation.categories.map((category) => (
```

Comment:
I think it would be beneficial to consider a design that clearly indicates that the checkbox cannot be clicked during the second rotation and onwards.